cluster/plan.go

```diff
@@ -173,24 +177,25 @@ func (c *Cluster) BuildKubeAPIProcess(host *hosts.Host, serviceOptions v3.Kubern
 
 	Command := c.getRKEToolsEntryPoint(host.OS(), "kube-apiserver")
 	CommandArgs := map[string]string{
-		"client-ca-file":               pki.GetCertPath(pki.CACertName),
-		"cloud-provider":               c.CloudProvider.Name,
-		"etcd-cafile":                  etcdCAClientCert,
-		"etcd-certfile":                etcdClientCert,
-		"etcd-keyfile":                 etcdClientKey,
-		"etcd-prefix":                  etcdPathPrefix,
-		"etcd-servers":                 etcdConnectionString,
-		"kubelet-client-certificate":   pki.GetCertPath(pki.KubeAPICertName),
-		"kubelet-client-key":           pki.GetKeyPath(pki.KubeAPICertName),
-		"proxy-client-cert-file":       pki.GetCertPath(pki.APIProxyClientCertName),
-		"proxy-client-key-file":        pki.GetKeyPath(pki.APIProxyClientCertName),
-		"requestheader-allowed-names":  pki.APIProxyClientCertName,
-		"requestheader-client-ca-file": pki.GetCertPath(pki.RequestHeaderCACertName),
-		"service-account-key-file":     pki.GetKeyPath(pki.ServiceAccountTokenKeyName),
-		"service-cluster-ip-range":     c.Services.KubeAPI.ServiceClusterIPRange,
-		"service-node-port-range":      c.Services.KubeAPI.ServiceNodePortRange,
-		"tls-cert-file":                pki.GetCertPath(pki.KubeAPICertName),
-		"tls-private-key-file":         pki.GetKeyPath(pki.KubeAPICertName),
+		"admission-control-config-file": DefaultKubeAPIArgAdmissionControlConfigFileValue,
```

Comment:
Will this not cause a redeploy of the `kube-apiserver` pod anytime someone upgrades to a version of RKE that contains this change?

Reply:
I am mostly concerned about this, as it has large implications for upgrades of Rancher.

Reply:
@Oats87 
Good point!
If  it is true that Rancher will run the reconciliation cycle against all downstream clusters after upgrading, I think, yes, the `kube-apiserve` pod will be redeployed. 
To me, it looks like something naturally happening based on how Rancher works today. I guess this is not the first we run into such a situation. Do you know any way to avoid redeployment? I also wonder if it is a good idea to add special logic to handle it in Rancher.

Reply:
update:
This is confirmed offline that Rancher does NOT run the reconciliation cycle against all downstream RKE clusters after upgrading. However, Rancher updates the plan on the worker nodes after upgrading. I need to makes sure that the downstream cluster is still working file in this case.